From 63b4227fa333db06cf7bc13fd9b333b6a9ec1f53 Mon Sep 17 00:00:00 2001
From: Andrew Bayer <andrew.bayer@gmail.com>
Date: Mon, 28 Feb 2011 11:13:56 -0800
Subject: [PATCH 14/51] CLOUDERA-BUILD. Running mvn to generate properties before anything else
 happens. Also only running it if we don't already have cdh.build.properties
 existing - no need to re-run in that case.

---
 build-common.xml          |   43 +++++++++++++++++++++++++++----------------
 build.xml                 |   33 +++++++++++++++++++++++----------
 cloudera/do-release-build |    3 ---
 3 files changed, 50 insertions(+), 29 deletions(-)

diff --git a/build-common.xml b/build-common.xml
index 07dcd36..d20ca23 100644
--- a/build-common.xml
+++ b/build-common.xml
@@ -23,6 +23,32 @@
          name="hivecommon" default="jar">
 
   <property name="hive.root" location="${basedir}/.."/>
+
+  <taskdef resource="net/sf/antcontrib/antcontrib.properties">
+    <classpath>
+      <pathelement location="${hive.root}/testlibs/ant-contrib-1.0b3.jar"/>
+    </classpath>
+  </taskdef>
+
+  <!-- Initializes ivy/libraries.properties and build.properties with CDH versions -->
+  <if>
+    <and>
+      <not>
+        <isset property="cdh.maven" />
+      </not>
+      <not>
+        <available file="${hive.root}/cdh.build.properties" />
+      </not>
+    </and>
+    <then>
+      <exec  executable="mvn" dir="${hive.root}" failonerror="true">
+        <arg value="-f" />
+        <arg value="cloudera-pom.xml" />
+        <arg value="process-resources"/>
+      </exec>
+    </then>
+  </if>
+
   <property file="${hive.root}/cdh.build.properties"/>
   <property file="${hive.root}/build.properties"/>
   <property file="${user.home}/build.properties" />
@@ -71,12 +97,6 @@
     <path refid="classpath"/>
   </path>
 
-  <taskdef resource="net/sf/antcontrib/antcontrib.properties">
-    <classpath>
-      <pathelement location="${hive.root}/testlibs/ant-contrib-1.0b3.jar"/>
-    </classpath>
-  </taskdef>
-
   <!-- IVY properties set here -->
   <property name="build.ivy.dir" location="${build.dir.hive}/ivy"/>
   <property name="build.ivy.lib.dir" location="${build.ivy.dir}/lib"/>
@@ -125,16 +145,7 @@
     </condition>
   </target>
 
-  <!-- Initializes ivy/libraries.properties and build.properties with CDH versions -->
-  <target name="init-cdh" unless="cdh.maven">
-    <exec executable="mvn" dir="${hive.root}" failonerror="true">
-      <arg value="-f" />
-      <arg value="cloudera-pom.xml" />
-      <arg value="process-resources"/>
-    </exec>
-  </target>
-
-  <target name="ivy-download" depends="ivy-init-dirs,init-cdh"
+  <target name="ivy-download" depends="ivy-init-dirs"
           description="To download ivy" unless="offline">
     <get src="${ivy_repo_url}" dest="${ivy.jar}" usetimestamp="true"/>
   </target>
diff --git a/build.xml b/build.xml
index 5b77be0..55534fc 100644
--- a/build.xml
+++ b/build.xml
@@ -20,6 +20,27 @@
 
 <project name="hive" default="jar" xmlns:artifact="urn:maven-artifact-ant" >
 
+  <!-- load ant-contrib tasks to get the "if" task. -->
+  <taskdef resource="net/sf/antcontrib/antcontrib.properties">
+    <classpath>
+      <pathelement location="${basedir}/testlibs/ant-contrib-1.0b3.jar"/>
+    </classpath>
+  </taskdef>
+
+  <!-- Initializes ivy/libraries.properties and build.properties with CDH versions -->
+  <if>
+    <not>
+      <isset property="cdh.maven" />
+    </not>
+    <then>
+      <exec  executable="mvn" dir="${basedir}" failonerror="true">
+        <arg value="-f" />
+        <arg value="cloudera-pom.xml" />
+        <arg value="process-resources"/>
+      </exec>
+    </then>
+  </if>
+
   <property name="hive.root" location="${basedir}"/>
   <property file="${hive.root}/cdh.build.properties"/>
   <property file="${hive.root}/build.properties"/>
@@ -161,16 +182,7 @@
     <path refid="common-classpath"/>
   </path>
 
-  <!-- Initializes ivy/libraries.properties and build.properties with CDH versions -->
-  <target  name="init-cdh" unless="cdh.maven">
-    <exec  executable="mvn" dir="${basedir}" failonerror="true">
-      <arg value="-f" />
-      <arg value="cloudera-pom.xml" />
-      <arg value="process-resources"/>
-    </exec>
-  </target>
-
-  <target name="init" depends="init-cdh">
+  <target name="init">
     <iterate-all target="init"/>
   </target>
 
@@ -252,6 +264,7 @@
     <iterate-all target="clean"/>
     <iterate-cpp target="clean"/>
     <delete dir="${target.dir}"/>
+    <delete file="${basedir}/cdh.build.properties" />
     <antcall target="clean-online"/>
     <antcall target="clean-offline"/>
   </target>
diff --git a/cloudera/do-release-build b/cloudera/do-release-build
index 9c2438c..ad79722 100755
--- a/cloudera/do-release-build
+++ b/cloudera/do-release-build
@@ -11,11 +11,8 @@ RELEASE_DIR=$BIN_DIR/..
 
 cd $RELEASE_DIR
 
-ant init-cdh
-
 mkdir -p build/.ivy
 ant $EXTRA_ANT_FLAGS \
-  -propertyfile cdh.build.properties \
   -propertyfile cloudera/build.properties \
   tar
 
-- 
1.7.0.4

